packages/tracing/src/transaction.ts

```diff
@@ -229,39 +230,35 @@ export class Transaction extends SpanClass implements TransactionInterface {
     const client = hub && hub.getClient();
 
     const { environment, release } = (client && client.getOptions()) || {};
-    const { publicKey } = (client && client.getDsn()) || {};
+    const { publicKey: public_key } = (client && client.getDsn()) || {};
```

Comment:
Can we try early returning if client is not defined? This might clean some of this up.

Reply:
Yup, in this case, I'll now return an empty baggage object but if the client is missing we have other problems than baggage anyway